samcli/commands/delete/delete_context.py

```diff
@@ -0,0 +1,137 @@
+"""
+Delete a SAM stack
+"""
+
+import boto3
+
+import click
+from click import confirm
+from click import prompt
+from samcli.cli.cli_config_file import TomlProvider
+from samcli.lib.utils.botoconfig import get_boto_config_with_user_agent
+from samcli.lib.delete.cf_utils import CfUtils
+from samcli.lib.package.s3_uploader import S3Uploader
+from samcli.lib.package.artifact_exporter import mktempfile, get_cf_template_name
+
+CONFIG_COMMAND = "deploy"
+CONFIG_SECTION = "parameters"
+TEMPLATE_STAGE = "Original"
+
+
+class DeleteContext:
+    def __init__(self, stack_name: str, region: str, profile: str, config_file: str, config_env: str):
+        self.stack_name = stack_name
+        self.region = region
+        self.profile = profile
+        self.config_file = config_file
+        self.config_env = config_env
+        self.s3_bucket = None
+        self.s3_prefix = None
+        self.cf_utils = None
+        self.s3_uploader = None
+        self.cf_template_file_name = None
+        self.delete_artifacts_folder = None
+        self.delete_cf_template_file = None
+
+    def __enter__(self):
+        self.parse_config_file()
+        if not self.stack_name:
+            self.stack_name = prompt(
+                click.style("\tEnter stack name you want to delete:", bold=True), type=click.STRING
+            )
+
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def parse_config_file(self):
+        """
+        Read the provided config file if it exists and assign the options values.
+        """
+        toml_provider = TomlProvider(CONFIG_SECTION, [CONFIG_COMMAND])
+        config_options = toml_provider(
+            config_path=self.config_file, config_env=self.config_env, cmd_names=[CONFIG_COMMAND]
+        )
+        if config_options:
+            if not self.stack_name:
+                self.stack_name = config_options.get("stack_name", None)
+            if self.stack_name == config_options["stack_name"]:
+                if not self.region:
+                    self.region = config_options.get("region", None)
+                if not self.profile:
+                    self.profile = config_options.get("profile", None)
```

Comment:
for region && profile also update SAM context, to refresh the default boto session